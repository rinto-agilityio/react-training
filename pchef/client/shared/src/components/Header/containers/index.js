import { compose } from 'react-apollo'

// graphQL query
import { getUser } from '../../../graphql/user'

// component
import Header from '../'

export default compose(getUser)(Header)
