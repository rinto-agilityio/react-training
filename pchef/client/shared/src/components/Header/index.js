// Libs
import React, { useState } from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import { Avatar, Provider, Menu } from 'react-native-paper'

// Common components
import Icon from '../Icon'
import Loading from '../Loading'
import Error from '../Error'
import Button from '../Button'

// Themes
import { COLORS, FONTS } from '../../themes'

// Styles
import styles from './styles'

// Constants
import { URL } from '../../constants/index'

type Props = {
  data: {
    user: {
      avatar: string,
    }
  },
  type?: string,
  onPressLogo?: () => void,
  onPressCategoryIcon?: () => void,
  onDirectTo: (id: string) => void,
}

const Header = ({
  type = 'primary',
  onPressLogo,
  onPressCategoryIcon,
  data = {
    user: {
      avatar: '',
    },
  },
  onDirectTo,
}: Props) => {
  const [visible, setVisible] = useState(false)
  const { user: { avatar } } = data

  return (
    <View style={[styles.wrapHeader, styles[`${type}Container`]]}>
      <View style={styles.container}>
        <Icon
          name="apps"
          size={FONTS.fontSize.extraExtraLarge}
          onPress={onPressCategoryIcon}
          color={COLORS.white}
          wrapperIconStyle={styles.wrapIcon}
        />
        <View style={styles.wrapHeaderContent}>
          <View>
            <TouchableOpacity onPress={onPressLogo} style={styles.logo}>
              <View style={[styles.imageWrapper, styles[`${type}ImageWrapper`]]}>
                <Image
                  source={require('../../assets/images/logo.png')}
                  style={[styles.image, styles[`${type}Image`]]}
                />
              </View>
              <Text style={styles.text}>Cooking</Text>
            </TouchableOpacity>
          </View>
          <View style={styles.wrapUserInfo}>
            <Button
              onPress={() => onDirectTo(URL.CREATE_RECIPE.PATH)}
              title={URL.CREATE_RECIPE.TITLE}
              buttonStyle={styles.createRecipeBtn}
            />
            <Provider>
              <Menu
                style={styles.avataDropdown}
                visible={visible}
                onDismiss={() => setVisible(false)}
                anchor={
                  (
                    <TouchableOpacity onPress={() => setVisible(true)} style={styles.logo}>
                      <Avatar.Image
                        style={styles.pipelineItemImage}
                        size={40}
                        source={{
                          uri: (
                            avatar
                              ? avatar
                              : 'https://www.pngkey.com/png/full/115-1150152_default-profile-picture-avatar-png-green.png'
                          ),
                        }}
                      />
                    </TouchableOpacity>
                  )
                }
              >
                <Menu.Item onPress={() => onDirectTo(URL.PROFILE.PATH)} title={URL.PROFILE.TITLE} />
                <Menu.Item onPress={() => onDirectTo(URL.WISH_LIST.PATH)} title={URL.WISH_LIST.TITLE} />
                <Menu.Item onPress={() => onDirectTo(URL.LOG_OUT.PATH)} title={URL.LOG_OUT.TITLE} />
              </Menu>
            </Provider>
          </View>
        </View>
      </View>
    </View>
  )
}

Header.defaultProps = {
  type: 'primary',
  onPressLogo: () => {},
  onPressCategoryIcon: () => {},
}

export default Header
