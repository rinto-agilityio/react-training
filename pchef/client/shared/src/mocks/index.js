// mock categories
export const category = {
  name: 'Heathy',
  imgUrl:
    'https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
}
