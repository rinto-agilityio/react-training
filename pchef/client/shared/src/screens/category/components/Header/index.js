// Libs
import React from 'react'
import { Text, View } from 'react-native'

// Styles
import styles from './styles'

// Themes
import { COLORS, METRICS } from '../../../../themes'

// Components
import ImageBackground from '../../../../components/ImageBackground'
import Icon from '../../../../components/Icon'
import Button from '../../../../components/Button'

type Props = {
  category: {
    name: string,
    imgUrl: string,
  },
  customTitle?: {},
  isGrid: boolean,
  customWrapperIcon?: {},
  size?: string,
  onFollowing?: () => void,
  onSelectListView?: () => void,
}

const Header = ({
  category,
  onFollowing,
  customTitle,
  isGrid,
  customWrapperIcon,
  onSelectListView,
  size = '',
}: Props) => {
  const { name, imgUrl } = category

  // Data render list view category
  const data = [
    {
      name: 'view-list',
      wrapperIconStyle: {
        marginRight: METRICS.extraLargeMargin,
      },
      color: isGrid ? COLORS.lighterGray : COLORS.baseGray,
    },
    {
      name: 'apps',
      wrapperIconStyle: {},
      color: isGrid ? COLORS.baseGray : COLORS.lighterGray,
    },
  ]

  return (
    <>
      <View style={[styles.wrapperHeader, styles[`${size}Header`]]}>
        <ImageBackground
          url={imgUrl}
          customImageBg={METRICS.flexCenter}
          resizeMode="cover"
        >
          <Text style={[styles.title, styles[`${size}Title`], customTitle]}>
            {name}
          </Text>
          <Button
            title="Following"
            type="outline"
            onPress={onFollowing}
            buttonStyle={styles.button}
            titleStyle={styles.titleBtn}
            size={size}
          />
        </ImageBackground>
      </View>
      <View style={[styles.wrapperHeaderIcon, customWrapperIcon]}>
        {data.map(item => (
          <Icon
            key={item.name}
            name={item.name}
            size={METRICS[`${size}Icon`]}
            color={item.color}
            wrapperIconStyle={item.wrapperIconStyle}
            onPress={onSelectListView}
          />
        ))}
      </View>
    </>
  )
}

Header.defaultProps = {
  customWrapperIcon: {},
  customTitle: {},
  size: 'small',
  onFollowing: () => {},
  onSelectListView: () => {},
}

export default Header
