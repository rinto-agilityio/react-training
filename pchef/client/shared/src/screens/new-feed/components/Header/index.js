// Libs
import React from 'react'
import { View, Text } from 'react-native'
import Wrapper from '../../../../layout/Wrapper'
import Icon from '../../../../components/Icon'
import Image from '../../../../components/Image'

// Themes
import { COLORS, FONTS } from '../../../../themes'

// Styles
import styles from './styles'

type Props = {
  image: string,
  onPressIcon?: () => void,
  size: string,
  handleTouch?: () => void,
}

const Header = ({
  image,
  onPressIcon,
  size = 'medium',
  handleTouch,
}: Props) => (
  <View
    style={[
      styles.contentHeader,
      styles[`${size}Content`],
    ]}
  >
    <Icon
      name="apps"
      size={size === 'large' ? FONTS.fontSize.extraExtraLarge : FONTS.fontSize.extraLarge}
      onClick={onPressIcon}
      color={COLORS.white}
      wrapperIconStyle={styles.wrapIcon}
    />
    <Wrapper direction="row">
      <Image
        url={image}
        customImageStyle={[
          styles.image,
          styles[`${size}Image`],
        ]}
        handleTouch={handleTouch}
      />
      <Text style={styles.text}>
        Cooking
      </Text>
    </Wrapper>
  </View>
)

Header.defaultProps = {
  onPressIcon: () => {},
  handleTouch: () => {},
}

export default Header
