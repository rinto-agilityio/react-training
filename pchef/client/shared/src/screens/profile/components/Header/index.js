import React from 'react'

// Components
import { View, Text } from 'react-native'
import Image from '../../../../components/Image'
import Button from '../../../../components/Button'

// Styles
import styles from './styles'

type Props = {
  user: {
    id: string,
    name: string,
    avatar: string,
  },
  handleToSetting?: () => void,
}

const Header = ({ user, handleToSetting }: Props) => {
  const { name, avatar } = user

  return (
    <View style={styles.wrapHeader}>
      <View style={styles.contentHeader}>
        <Image url={avatar} customImageStyle={styles.image} />
      </View>
      <View style={styles.user}>
        <Text style={styles.userName}>{name}</Text>
        <Button
          buttonStyle={styles.userButton}
          titleStyle={styles.userTitleButton}
          title="Edit profile"
          onPress={handleToSetting}
        />
      </View>
    </View>
  )
}

Header.defaultProps = {
  handleToSetting: () => {},
}

export default Header
