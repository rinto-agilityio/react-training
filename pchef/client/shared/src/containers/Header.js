import { compose } from 'react-apollo'

// graphQL query
import { getUser } from '../graphql/user'

// component
import Header from '../components/Header'

export default compose(getUser)(Header)
