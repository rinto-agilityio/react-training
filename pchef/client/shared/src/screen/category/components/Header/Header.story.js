// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Header from '.'

// Mock
import { categories } from '../../../../mocks'

storiesOf('Header', module).add('default', () => (
  <Header category={categories[0]} isGrid sizeType="large" />
))
