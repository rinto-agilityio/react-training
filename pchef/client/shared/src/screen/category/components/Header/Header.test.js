// Libs
import React from 'react'
import renderer from 'react-test-renderer'

// Components
import Header from '.'

// Mock
import { category } from '../../../../mocks'

const props = {
  category,
}

describe('Components', () => {
  describe('Header', () => {
    let component

    beforeEach(() => {
      component = shallow(<Header {...props} />)
    })

    it('should render Header component with grid', () => {
      const renderComponent = renderer.create(<Header {...props} isGrid />)
      expect(renderComponent).toMatchSnapshot()
    })

    it('should render Header component without grid', () => {
      expect(
        component
          .find('Icon')
          .at(0)
          .props().color,
      ).toEqual('#c5c5c5')
    })
  })

  it('should have default onFollowing props', () => {
    Header.defaultProps.onFollowing()
    expect(Header.defaultProps.onFollowing).toBeDefined()
  })

  it('should have default onSelectListView props', () => {
    Header.defaultProps.onSelectListView()
    expect(Header.defaultProps.onSelectListView).toBeDefined()
  })
})
