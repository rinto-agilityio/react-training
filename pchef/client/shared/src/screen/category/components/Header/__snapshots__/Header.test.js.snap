// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header should render Header component with grid 1`] = `
Array [
  <div
    className="css-view-1dbjc4n r-display-6koalj r-height-1kl1mp1"
  >
    <div
      className="css-view-1dbjc4n r-height-1pi2tsx r-width-13qz1uu"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxPack": "center",
          "WebkitJustifyContent": "center",
          "alignItems": "center",
          "justifyContent": "center",
          "msFlexAlign": "center",
          "msFlexPack": "center",
        }
      }
    >
      <div
        className="css-view-1dbjc4n r-bottom-1p0dtai r-flexBasis-1mlwlqe r-left-1d2f490 r-overflow-1udh08x r-position-u8s1d r-right-zchlnj r-top-ipm5af r-zIndex-1wyyakw"
      >
        <div
          className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
          style={
            Object {
              "backgroundImage": "url(\\"https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg\\")",
            }
          }
        />
        <img
          alt=""
          className="css-accessibilityImage-9pa8cd"
          draggable={false}
          src="https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg"
        />
      </div>
      <div
        className="css-text-76zvg2 r-color-jwli3a r-fontSize-1i10wst r-fontWeight-1kfrs79 r-marginBottom-d0pm55"
        dir="auto"
      >
        Chicken
      </div>
      <div
        className="css-view-1dbjc4n r-backgroundColor-1niwhzg"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxPack": "center",
            "WebkitJustifyContent": "center",
            "alignItems": "center",
            "borderBottomColor": "rgba(255,255,255,1.00)",
            "borderBottomLeftRadius": "20px",
            "borderBottomRightRadius": "20px",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "2px",
            "borderLeftColor": "rgba(255,255,255,1.00)",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "2px",
            "borderRightColor": "rgba(255,255,255,1.00)",
            "borderRightStyle": "solid",
            "borderRightWidth": "2px",
            "borderTopColor": "rgba(255,255,255,1.00)",
            "borderTopLeftRadius": "20px",
            "borderTopRightRadius": "20px",
            "borderTopStyle": "solid",
            "borderTopWidth": "2px",
            "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
            "justifyContent": "center",
            "minWidth": "64px",
            "msFlexAlign": "center",
            "msFlexPack": "center",
            "paddingBottom": "5px",
            "paddingLeft": "20px",
            "paddingRight": "20px",
            "paddingTop": "5px",
          }
        }
      >
        <div
          className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-overflow-1udh08x r-touchAction-1otgn73 r-userSelect-lrvibr"
          data-focusable={true}
          disabled={false}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="button"
          style={
            Object {
              "borderBottomLeftRadius": "4px",
              "borderBottomRightRadius": "4px",
              "borderTopLeftRadius": "4px",
              "borderTopRightRadius": "4px",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci"
          >
            <div
              className="css-text-76zvg2 css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
              dir="auto"
              style={
                Object {
                  "color": "rgba(0,0,0,1.00)",
                  "direction": "ltr",
                  "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                }
              }
            >
              <span
                className="css-text-76zvg2 css-textHasAncestor-16my406 r-color-jwli3a r-fontSize-1b43r93 r-fontWeight-1kfrs79 r-textTransform-tsynxw"
                dir="auto"
                style={
                  Object {
                    "direction": "ltr",
                    "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                    "textAlign": "left",
                  }
                }
              >
                Following
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-4rzz9f r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1777fci r-padding-1sp51qo"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
      style={
        Object {
          "marginRight": "30px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="css-hitSlop-mjp8i1"
          style={
            Object {
              "bottom": "-6px",
              "left": "-6px",
              "right": "-6px",
              "top": "-6px",
            }
          }
        />
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-76zvg2 r-backgroundColor-1niwhzg r-pointerEvents-633pao"
            dir="auto"
            focusable={false}
            role="img"
            style={
              Object {
                "color": "rgba(31,51,64,1.00)",
                "fontSize": "20px",
              }
            }
          >
            □
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="css-hitSlop-mjp8i1"
          style={
            Object {
              "bottom": "-6px",
              "left": "-6px",
              "right": "-6px",
              "top": "-6px",
            }
          }
        />
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-76zvg2 r-backgroundColor-1niwhzg r-pointerEvents-633pao"
            dir="auto"
            focusable={false}
            role="img"
            style={
              Object {
                "color": "rgba(197,197,197,1.00)",
                "fontSize": "20px",
              }
            }
          >
            □
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
